### PR TITLE
Simple addition of a new view 'Context View'

### PR DESCRIPTION
**Pull request by PAHXO**
Hello this is a different approach to the normal view

It still needs some design changes regarding the area near the date/link side

Changes proposed in this pull request:

- Adding context view

Pull request checklist:

- [ ] adding localization and some icons
- [x] ```timplate.css/rtl``` needs more cleaning  [link](https://travis-ci.org/github/FreshRSS/FreshRSS/jobs/754026118#L235)
- [ ] (Icon needed) ``` /theme/icons/view-contextview.svg``` // filename may require a change
- [ ] (Image needed) ```/theme/icons/article-no-picture.png``` // filename may require a change
- [ ] localization ``` 'contextview' => 'Contextview view'``` in  ```i18n\*\conf.php```
- [ ] localization ``` 'contextview_view' => 'Contextview view' ```in  ```i18n\*\index.php```
- [ ] moving the extension into the FreshRSS code
- [ ] documentation updated

[Additional information can be found in the documentation](https://github.com/FreshRSS/FreshRSS/tree/master/docs/en/developers/04_Pull_requests.md).

### CODE REVIEW

File: app/views/helpers/index/contextview/entry_header.phtml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+				foreach ($authors as $author) {
+					echo $first ? $author : ', ' . $author;
+					$first = false;
+				}
+			}
+			?></span><?php
+		endif;
+	?>
+	</a>
+		<p class="item context" dir="auto"><SUP>&nbsp;</SUP><?= ContextViewExtension::addABriefInEntryHeader($this->entry); ?></p>
```
Comment (Alkarex) [inline app/views/helpers/index/contextview/entry_header.phtml]: `<SUP>&nbsp;</SUP>` what are you trying to achieve?

File: app/views/helpers/index/contextview/entry_header.phtml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	</li>
+	<?php if (true) { ?><li class="item image"> <a target="_blank" dir=auto rel="noreferrer" href="<?= $this->entry->link() ?>" <time datetime="<?= $this->entry->machineReadableDate() ?>"> <?=ContextViewExtension::humanTiming(strtotime($this->entry->machineReadableDate())).'';?> <?=ContextViewExtension::imageInEntryHeader($this->entry);?> </time></a></li><?php } ?>
```
Comment (Alkarex) [inline app/views/helpers/index/contextview/entry_header.phtml]: `if (true)` most likely not needed

File: app/views/helpers/index/contextview/entry_bottom.phtml [hunk 3]
```diff
@@ -0,0 +1,115 @@
+<?php
+	$sharing = array();
```
Comment (Alkarex) [inline app/views/helpers/index/contextview/entry_bottom.phtml:2]: No need to change it, but for future code, please favour the new `= []` syntax

File: app/views/helpers/index/contextview/entry_header.phtml (hunk 2 above)
Comment (PAHXO) [reply app/views/helpers/index/contextview/entry_header.phtml]: I know I said it is a placeholder for a condition if needed.

File: app/views/helpers/index/contextview/entry_header.phtml (hunk 1 above)
Comment (PAHXO) [reply app/views/helpers/index/contextview/entry_header.phtml]: ```<br>```  is too high I read that this is a half line.

**Comment by Alkarex** [comment]
Thanks! Can you please post here a few screenshots to ease reviewing from several persons?

**Comment by PAHXO** [comment]
Screenshot 2021-01-12 011136

RTL test
Screenshot 2021-01-12 011225

**Comment by Alkarex** [comment]
Ah, there was already a screenhost in https://github.com/FreshRSS/FreshRSS/issues/3135 but it is better to have some here as well :-)

File: app/Models/ConfigurationSetter.php [hunk 4]
```diff
@@ -149,7 +149,7 @@ private function _ttl_default(&$data, $value) {
 
 	private function _view_mode(&$data, $value) {
 		$value = strtolower($value);
-		if (!in_array($value, array('global', 'normal', 'reader'))) {
+		if (!in_array($value, array('global', 'normal', 'reader', 'contextview'))) {
```
Comment (Alkarex) [inline app/Models/ConfigurationSetter.php:152]: I would better like a name that does not end with -view, to be closer to the other names. *context* might be fine, but maybe we can find something better. What is used in other software proposing such a view? Maybe *preview*, *timeline*, *magazine*...

Comment (PAHXO) [reply app/Models/ConfigurationSetter.php:152]: Yeah, I agree.

File: app/views/helpers/index/contextview/entry_header.phtml (hunk 2 above)
Comment (aledeg) [reply app/views/helpers/index/contextview/entry_header.phtml]: If we don't need it now, there is no need to have it at all. It the need arise, we will add it!

Comment (PAHXO) [inline app/views/helpers/index/contextview/entry_header.phtml]:
```suggestion
	<li class="item image">
	<a target="_blank" dir=auto rel="noreferrer" href="<?= $this->entry->link() ?>" 
	<time datetime="<?= $this->entry->machineReadableDate() ?>">
	<?=ContextViewExtension::humanTiming(strtotime($this->entry->machineReadableDate())).''?> 
	</time>
	<?=ContextViewExtension::imageInEntryHeader($this->entry);?>
	</a></li>
```

**Review by PAHXO — COMMENTED** [review]
It should be a bit better like this I guess. Sorry, I'm still very new to GitHub etiquette. a couple of weeks ago I had zero knowledge about PHP.

File: app/views/helpers/index/contextview/entry_header.phtml (hunk 1 above)
Comment (Alkarex) [reply app/views/helpers/index/contextview/entry_header.phtml]: `<sup>` is to make text as exponent, like x². If I understand what you want, this should be done by CSS instead.

**Comment by Alkarex** [comment]
@PAHXO I have resolved some recent merge conflicts, and fixed some CSS/JS warning/errors. Please double-check. Let us know how you progress. That looks good so far

**Comment by PAHXO** [comment]
Looks good thanks!

**Comment by PAHXO** [comment]
Done a few improvements. 
Screenshot 2021-03-10 114657
Screenshot 2021-03-10 114809

**Comment by Alkarex** [comment]
@PAHXO That looks very good 👍🏻

**Comment by PAHXO** [comment]
Closing this pr.
Thanks to Orelio's #3805 it was done better. Sorry, that I left this pr, but I'm not so comfortable in coding reliable stuff and having a clean code.
Anyway, I have experimented some stuff related to  #2402, yet not so similar.
I've been using this for a while now, and it feels really nice on a large screen.


Untitled

Untitled
just a very simple CSS lines added using the grid approach.
Sure, it may not suit a lot of people, but I feel like this “context view or whatever it may be” takes a huge screen space for a very limited number of entries, which makes you miss some entries. You might have wished to notice…

**Comment by ORelio** [comment]
>  Thanks to Orelio's #3805 it was done better. Sorry, that I left this pr, but I'm not so comfortable in coding reliable stuff and having a clean code.

Sorry for not seeing your PR before reimplementing it 😨
The "Cards" view looks nice too 👍

**Comment by PAHXO** [comment]
> @ORelio 
> The "Cards" view looks nice too 👍


Thanks! It's not much just tho, but it can easily be combined with the “thumbnail and summary” view to look a lot better.

```
@media screen and (min-width: 1400px) {
	.normal main#stream {
		display: grid;
		grid-column-gap: 10px;
		grid-template-columns: auto auto auto;
		padding: 10px;
	}


	.normal .flux .item.website {
	/* display: none; */
                 width: 25px;
	}
	
	.normal .flux .item {
		margin-left: 10%;
		line-height: 25px;
	}
	.normal .flux .item.title a {
		margin-left: 8%;
	}
		
	div#day_today {
		/*display: none;*/
	}
	div#day_yesterday {
		display: none;
	}
	div#day_before_yesterday {
		display: none;
	}	
	
	.normal .flux .item.context {
		text-align: start;
		margin-left: 8%;
		margin-right: 8%;
	}
	.normal .flux  .item.title a {
		text-align: start;
		margin-left: 8%;
		margin-right: 8%;
	}
}
```

**Pull request closed without merging**